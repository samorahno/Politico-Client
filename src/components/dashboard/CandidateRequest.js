import React from 'react';
import '../../styles/dashboard.css';
import Header from '../nav/DashboardHeader';
import Footer from '../nav/DashboardFooter';
import SideBar from '../nav/DashboardSidebar';

const CandidateRequest = () => {
    return (
        <div>
            <Header />
            <SideBar />
            <div className="main-content">
                <div className="title">
					&nbsp;
                </div>

                <div className="main">

                    <div className="widget widget_table" style={{ overflow: 'auto' }}>
                        <div className="title heado">Candidate Request List</div>

                        <div className="widget_stat widget_stat_table">
                            <div style={{ overflow: 'auto' }}>
                                <table style={{ overflow: 'scroll' }}>
                                    <thead >
                                        <tr>
                                            <th>S/N</th>
                                            <th>Candidate</th>
                                            <th>Office</th>
                                            <th>Date Created</th>
                                            <th>Action</th>
                                        </tr>
                                    </thead>

                                    <tbody id="interestContainer">
                                        <tr>
                                            <td>1</td>
                                            <td>A</td>
                                            <td>B</td>
                                            <td>C</td>
                                            <td>D</td>
                                        </tr>
                                        <tr>
                                            <td>2</td>
                                            <td>A</td>
                                            <td>B</td>
                                            <td>C</td>
                                            <td>D</td>
                                        </tr>
                                    </tbody>

                                </table>
                            </div>
                        </div>
                    </div>

                </div>
            </div>
            <Footer />
        </div>

    );
};

export default CandidateRequest;
