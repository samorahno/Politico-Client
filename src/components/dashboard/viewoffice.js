import React, { Component } from 'react';


class ViewOffice extends Component {
    render () {
        return (
            <div>
                <h1>&nbsp;</h1>
                <h1>This is the admin office, You are logged in.........</h1>
            </div>
        );
    }
}

export default ViewOffice;
