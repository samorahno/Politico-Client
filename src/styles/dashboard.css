* {
	box-sizing: border-box;
}

body {
	margin: 0px;
	padding: 0px;
	background-color: #5f9ea0;
	font-family: 'Source Sans Pro', sans-serif;
	-webkit-font-smoothing: antialiased;
	-moz-osx-font-smoothing: grayscale;
}

body a {
	text-decoration: none;
}

table th{
	background-color: #EFBFA8;
}
header {
	position: absolute;
	height: 44px;
	width: 100%;
	background-color: #5f9ea0;
	color: white;
	z-index: 3;
	margin-top: 0;
}

header a {
	color: white;
	font-size: 30px;
	text-decoration: none;
	padding-left: 20px;
	font-weight: bold;
}

header a.minimize_button {
	float: right;
	padding-right: 15px;
}

.minimize_button{
	display: block;
}

.side-nav {
	position: absolute;
	height: 100%;
	width: 100%;
	background-color: #5f9ea0;
	z-index: 1;
	padding-top: 15px;
}
.side-nav visible {
	display: block;
}

.side-nav.visible {
	display: none;
}

.side-nav ul {
	list-style-type: none;
	margin-top: 30px;
	padding: 0px;
}

.side-nav ul li {
	padding: 10px 10px;
	border-bottom: 1px solid white;
	background-color: #5f9ea0;
	margin-top:10px;
}

.side-nav ul li a {
	text-decoration: none;
	color: white;
}

.side-nav ul li.active {
	background-color: gray;
}

.side-nav ul a.active{
	background-color: gray;
	padding: 10px 10px;
}

.side-nav ul li span i {
	text-decoration: none;
	color: black;
	font-size: 22px;
}

/* main content styles */

.main-content {
	margin-top: 0;
	margin-left: 0;
	padding: 0;
	padding-top: 30px;
	background-color: #d5dae5;
}

.main-content .title {
	background-color: #eef1f7;
	border-bottom: 1px solid #b8bec9;
	padding: 10px 20px;
	font-weight: 700;
	color: #333;
	font-size: 18px;
	width: 100%;
	margin-right: auto;
	margin-left: auto;
	margin-top: 20px;
}

@media screen and (min-width:600px) {
	.side-nav.visible {
		display: block;
	}
	.side-nav {
		width: 80px;
		height: 90vh;
	}
	.side-nav ul li {
		text-align: center;
	}
	.side-nav ul li span:nth-child(2) {
		display: none;
	}
	.side-nav ul li span i {
		font-size: 26px;
	}
	header a.minimize_button {
		display: none;
	}
	.main-content {
		margin-left: 70px;
		margin-top: 0;
	}
}

@media screen and (min-width:800px) {
	.minimize_button{
		display: block;
	}
	.side-nav visible {
		display: block;
	}
	.side-nav {
		width: 195px;
		height: 50%;
		margin-top: 100px;
	}
	.side-nav ul li {
		text-align: left;
	}
	.side-nav ul li span:nth-child(2) {
		display: inline;
	}
	.side-nav ul li span i {
		font-size: 26px;
	}
	header a.minimize_button {
		display: none;
	}
	.main {
		display: flex;
		flex-flow: row wrap;
	}
	.main-content {
		margin-left: 195px;
		height: 100%;
		background-color: #d5dae5;
		min-height: 700px;
	}
	.widget {
		flex-basis: 100px;
		flex-grow: 10;
		height: 200px;
		margin: 15px;
		border-radius: 6px;
		background-color: #fff;
		position: relative;
	}
}

.main {
	display: flex;
	flex-flow: row wrap;
}

.widget {
	flex-basis: 100px;
	flex-grow: 10;
	height: 200px;
	margin: 15px;
	border-radius: 6px;
	background-color: #fff;
	position: relative;
}

.widget.formval {
	flex-basis: 200px;
	flex-grow: 10;
	height: auto;
	margin: 15px;
	border-radius: 6px;
	background-color: #fff;
	position: relative;
}

.widget .title {
	background-color: #eef1f7;
	border-bottom: 1px solid #dfe4ec;
	padding: 10px;
	border-top-left-radius: 6px;
	border-top-right-radius: 6px;
	color: #617085;
	font-weight: 600;
}

.widget_stat {
	text-align: center;
	line-height: 40px;
	display: block;
}

.widget_stat1 {
	font-size: 40px;
	margin-top: 15px;
}

.widget_stat2 {
	font-size: 50px;
	font-weight: bolder;
	margin: 8px;
}

footer {
	background-color: #5f9ea0;
	border-bottom: 1px solid #5f9ea0;
	padding: 10px 20px;
	font-weight: 700;
	color: #fff;
	font-size: 18px;
	width: 100%;
	text-align: center;
	margin-bottom: 0;
}

form {
	background-color: white;
	text-align: center;
	width: 80%;
	margin-right: auto;
	margin-left: auto;
	padding: 10px;
}

form input {
	margin: 10px;
	border-radius: 5px;
	font-size: 15px;
	height: 40px;
	padding: 10px;
	width: 60%;
	border: 1px solid gray;
}

form textarea {
	margin: 10px;
	border-radius: 5px;
	font-size: 15px;
	height: 70px;
	padding: 10px;
	width: 60%;
	border: 1px solid gray;
}

form select {
	margin: 10px;
	border-radius: 5px;
	font-size: 15px;
	height: 40px;
	padding: 10px;
	width: 60%;
	border: 1px solid gray;
}

.input-group {
	margin: 10px 0px 10px 0px;
}

.input-group label {
	display: inline;
	text-align: left;
	margin: 3px;
	padding-left: 20px;
}

.btn {
	border-radius: 5px;
	padding: 10px;
	color: white;
	font-size: 15px;
	display: inline;

	margin-right: auto;
	margin-left: auto;
	background: #5f9ea0;
}

.title.heado {
	margin-top: 0;
}

table {
	text-align: center;
	width: 100%;
}

.overflo {
	overflow-x: auto;
}

.votesize {
	width: 50%;
}

table tbody tr td {
	border-bottom: 1px solid black;
}

.widget.widget_table {
	flex-basis: 400px;
	height: auto;
}

.widget_stat.widget_stat_table {
	text-align: center;
	width: 100%;
}

.usertable tr:nth-child(even) {
	background-color: #eef1f7;
}

#green {
	color: green;
}

#orange {
	color: orange;
}

#red {
	color: red;
}

button {
	cursor: pointer;
}

.btn-round {
	border-radius: 6px;
}

.tablelink {
	font-weight: bolder;
	color: black;
}

.partylogo {
	width: 150px;
	height: 150px;
	border-radius: 7px;
}

.voteresult {
	list-style-type: none;
}

.btnmore {
	width: 200px;
	margin: 10px;
}

.paddmore {
	margin:10px;
	width: 200px;
	margin-left: 35px;
}

.cards {
	display: flex;
	flex-wrap: wrap;
	align-items: stretch;
	padding-left:30px;
}

.card {
	flex: 0 0 200px;
	margin: 10px;
	border: 1px solid #ccc;
	box-shadow: 2px 2px 6px 0px rgba(0, 0, 0, 0.3);
	background-color: white;
	
}


.card img {
	max-width: 100%;
	text-align: center;
}

.card .text {
	padding: 0 20px 20px;
}

.card .text>button {
	background: gray;
	border: 0;
	color: white;
	padding: 10px;
	width: 100%;
}

.btn2 {

	background: #333;
	color: #fff;
	text-decoration: none;
	padding: 2px;
	border: 1px solid #666;
	margin: 0;
	width: 70px;
	border-radius:7px;
}

.info-msg,
.success-msg,
.warning-msg,
.error-msg {
  margin: 10px 0;
  padding: 10px;
  border-radius: 3px 3px 3px 3px;
  display: none;
  margin: 10px;
}
.info-msg {
  color: #059;
  background-color: #BEF;
}
.success-msg {
  color: #270;
  background-color: #DFF2BF;
  width: 40%;
  margin-right:auto;
  margin-left:auto;
}
.warning-msg {
  color: #9F6000;
  background-color: #FEEFB3;
}
.error-msg {
  color: #D8000C;
  background-color: #FFBABA;
  width: 40%;
  margin-right:auto;
  margin-left:auto;
}

#loader-overlay{
    height: 100%;
    width: 100%;
    background: #5f9ea0;
    position: fixed;
    left: 0;
    top: 0px;
}
.loader{
    margin: 15em auto;
    border: 16px solid #e3e3e3;
    border-radius: 50%;
    border-top: 16px solid #3498db;
    width: 100px;
    height: 100px;
    animation: spin 2s linear infinite;
    
}

@keyframes spin{
    0%{
        transform: rotate(0deg);
    }
    100%{
        transform: rotate(360deg);
    }
}

.toast-message {
  text-align: center;
}