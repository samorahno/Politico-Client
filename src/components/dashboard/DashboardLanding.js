import React from 'react';
import '../../styles/dashboard.css';
import Header from '../nav/DashboardHeader';
import Footer from '../nav/DashboardFooter'

const DashboardLanding = () => {
    return (
			<div>
				<Header />
        <div className='main-content'>
				<div className='title'>
					home /
				</div>
			
			<div className='main'>
				<div className='widget'>
					<div className='title heado'>Citizens</div>
					<div className='widget_stat'>
						<div className='widget_stat1'>
						<i className='fa  fa-check-circle-o'></i>
						</div>
						<div className='widget_stat2'>
						5
						</div>
					</div>
				</div>
				<div className='widget'>
					<div className='title heado'>Politicians</div>
					<div className='widget_stat'>
						<div className='widget_stat1'>
						<i className='fa fa-asterisk'></i>
						</div>
						<div className='widget_stat2'>
						8
						</div>
					</div>
				</div>
				<div className='widget'>
					<div className='title heado'>Vote Count</div>
					<div className='widget_stat'>
						<div className='widget_stat1'>
						<i className='fa fa-book'></i>
						</div>
						<div className='widget_stat2'>
						8
						</div>
					</div>
				</div>
				
			</div>
			
			<main className='cards'>
				  <article className='card'>
					<img src='../images/Anonymous.jpg' alt='Candidate Photo' />
					<div className='text'>
					  <h4><strong>Name:</strong> Wale Ojo</h4>
					  <p><strong>Office:</strong> National Assembly Member</p>
					  <p><strong>Party:</strong> PDP</p>
					 
					</div>
				  </article>
				  <article className='card'>
					<img src='../images/Anonymous.jpg' alt='Sample photo' />
					<div className='text'>
					  <h4><strong>Name:</strong> Ahmed Mark</h4>
					  <p><strong>Office:</strong> National Assembly Member</p>
					  <p><strong>Party:</strong> APC</p>
					 
					</div>
				  </article>
				  <article className='card'>
					<img src='../images/Anonymous.jpg' alt='Sample photo' />
					<div className='text'>
					  <h4><strong>Name:</strong> Bimbo Ali</h4>
					  <p><strong>Office:</strong> National Assembly Member</p>
					  <p><strong>Party:</strong> AD</p>
					
					</div>
				  </article>
				  <article className='card'>
					<img src='../images/Anonymous.jpg' alt='Sample photo' />
					<div className='text'>
					  <h4><strong>Name:</strong> Tolu Oni</h4>
					  <p><strong>Office:</strong> National Assembly Member</p>
					  <p><strong>Party:</strong> LP</p>
					 
					</div>
				  </article>
				  <article className='card'>
					<img src='../images/Anonymous.jpg' alt='Sample photo' />
					<div className='text'>
					  <h4><strong>Name:</strong> Tola Odeyemi</h4>
					  <p><strong>Office:</strong> Legislative Council Member</p>
					  <p><strong>Party:</strong> PDP</p>
					
					</div>
				  </article>
				  
        </main>
			<button className='btn btnmore'>View More</button>
			</div>
       <Footer />
			 </div>     
    );
}

export default DashboardLanding;