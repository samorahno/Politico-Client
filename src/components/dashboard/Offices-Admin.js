import React, { Component } from 'react';
import '../../styles/dashboard.css';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import { createOffice, getOffice } from '../../actions';
import Header from '../nav/DashboardHeader';
import Footer from '../nav/DashboardFooter';
import SideBar from '../nav/DashboardSidebar';
import ToastMessage from '../common/ToastMessage';
import ListOffices from '../../components/common/ListOffices';


class OfficeAdmin extends Component {
    constructor (props) {
        super(props);
        this.state = {
            officename: '',
            type: ''
        };
        this.modal = React.createRef();
        this.onChange = this.onChange.bind(this);
        this.onSubmit = this.onSubmit.bind(this);
    }

    componentDidMount () {
        const { listOffices } = this.props;
        listOffices();
    }
    onChange (e) {
        this.setState({ [e.target.name]: e.target.value });
    }

    onSubmit (e) {
        e.preventDefault();
        const { createNewOffice, createOfficeError, status, errorMessage } = this.props;
        const officeData = {
            name: this.state.officename,
            type: this.state.type
        };

        createNewOffice(officeData);
        if (createOfficeError && status >= 400) {
            toast(<ToastMessage message={errorMessage} />, {
                type: 'error',
                closeButton: false,
                hideProgressBar: true,
                autoClose: 5000
            });
        }

        if (!createOfficeError && status === 200) {
            toast(<ToastMessage message="Office successfully created" />, {
                type: 'success',
                closeButton: false,
                hideProgressBar: true,
                autoClose: 0
            });
        }
    }

    render () {
        const { isAdmin, offices } = this.props;
        return (
            <div>
                <Header />
                <SideBar />
                <div className="main-content">
                    <div className="title">
                        office /
                    </div>

                    {isAdmin && (
                        <div className="main">
                            <div className="widget widget_table" style={{ overflow: 'auto' }}>
                                <div className="title heado">Create New Political Office</div>
                                <div className="widget_stat widget_stat_table">
                                    <div className = "success-msg" id = "success-msg" />
                                    <div className = "warning-msg" id = "warning-msg" />
                                    <div className = "error-msg" id = "error-msg" />
                                    <div className = "info-msg" id = "info-msg" />
                                    <div style={{ overflow: 'auto' }}>
                                        <form action="#" method="GET" onSubmit={this.onSubmit}>

                                            <div className="input-group">

                                                <label>Office Name</label>
                                                <input
                                                    type="text"
                                                    name="officename"
                                                    placeholder="Office Name"
                                                    id="officeName"
                                                    onChange={this.onChange}
                                                    value={this.state.officename}
                                                />
                                            </div>
                                            <div className="input-group">
                                                <label>Office Type</label>
                                                <select id="officeType" name="type" onChange={this.onChange} value={this.state.value}>
                                                    <option value="">Select office type</option>
                                                    <option value="federal">Federal Government</option>
                                                    <option value="legislative">Legislative Government</option>
                                                    <option value="state">State Government</option>
                                                </select>
                                            </div>

                                            <div className="input-group">
                                                <label>&nbsp;&nbsp;&nbsp;
                                                &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
                                                &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
                                                &nbsp;&nbsp;</label>
                                                <button
                                                    type="submit"
                                                    name="submit"
                                                    className="btn"
                                                    id="officeBtn"
                                                >Create Office</button>
                                            </div>

                                        </form>
                                    </div>
                                </div>
                            </div>
                        </div>
                    )}


                    <div className="main">
                        <ListOffices offices={offices}/>
                    </div>
                </div>
                <Footer />
            </div>

        );
    }
}

const mapDispatchToProps = {
    createNewOffice: createOffice,
    listOffices: getOffice
};

const mapStateToProps = (state) => ({
    offices: state.office.office,
    createOfficeError: state.office.createOfficeError,
    status: state.office.status,
    errorMessage: state.office.error
});
export default connect(mapStateToProps, mapDispatchToProps)(OfficeAdmin);
