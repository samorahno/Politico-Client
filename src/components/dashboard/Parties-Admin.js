import React, { Component, Fragment } from 'react';
import '../../styles/dashboard.css';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import { createParty, getParty } from '../../actions';
import Header from '../nav/DashboardHeader';
import Footer from '../nav/DashboardFooter';
import SideBar from '../nav/DashboardSidebar';
import ToastMessage from '../common/ToastMessage';
import PartyCard from '../common/partiesCard';
import apcLogo from '../../images/apclogo.jpg';


class PartiesAdmin extends Component {
    constructor (props) {
        super(props);
        this.state = {
        };
        this.onChange = this.onChange.bind(this);
        this.onSubmit = this.onSubmit.bind(this);
    }
    componentDidMount () {
        const { listParties } = this.props;
        listParties();
    }
    componentDidUpdate (prevProps) {
        const { createPartyError, status, errorMessage } = this.props;
        if (prevProps.createPartyError !== createPartyError) {
            if (status >= 400) {
                toast(<ToastMessage message={errorMessage} />, {
                    type: 'error',
                    closeButton: false,
                    hideProgressBar: true,
                    autoClose: 5000
                });
            }
            if (!createPartyError) {
                toast(<ToastMessage message="Party successfully created" />, {
                    type: 'success',
                    closeButton: false,
                    hideProgressBar: true,
                    autoClose: 0
                });
            }
        }
    }
    onChange (e) {
        this.setState({ [e.target.name]: e.target.value });
    }

    onSubmit (e) {
        e.preventDefault();
        const { createNewParty } = this.props;
        const partyData = {
            name: this.state.partyname,
            alias: this.state.alias,
            hqAddress: this.state.partyaddress,
            logoUrl: this.state.partyLogo
        };

        createNewParty(partyData);
    }
    render () {
        const { parties, isAdmin } = this.props;
        return (
            <div>
                <Header />
                <SideBar />
                <div className="main-content">
                    <div className="title">
                        parties /
                    </div>

                    {isAdmin && (
                        <div className="main">


                            <div className="widget widget_table" style={{ overflow: 'auto' }}>
                                <div className="title heado">Create New Political Party</div>
                                <div className="widget_stat widget_stat_table">
                                    <div style={{ overflow: 'auto' }}>
                                        <form onSubmit={this.onSubmit}>

                                            <div className="input-group">

                                                <label>Party Name</label>
                                                <input
                                                    type="text"
                                                    name="partyname"
                                                    placeholder="Party Name"
                                                    id="partyName"
                                                    onChange={this.onChange}
                                                    value={this.state.partyname}
                                                />

                                            </div>
                                            <div className="input-group">

                                                <label>Abbreviation</label>
                                                <input
                                                    type="text"
                                                    name="alias"
                                                    placeholder="Party Initials (APC,PDP)"
                                                    id="partyAbbreviation"
                                                    onChange={this.onChange}
                                                    value={this.state.alias}
                                                />

                                            </div>
                                            <div className="input-group">

                                                <label>HQ Address</label>
                                                <input
                                                    type="text"
                                                    name="partyaddress"
                                                    placeholder="Head Quarters Address"
                                                    id="hqAddress"
                                                    onChange={this.onChange}
                                                    value={this.state.partyaddress}
                                                />

                                            </div>
                                            <div className="input-group">
                                                <label>Party Logo</label>
                                                <input
                                                    type="text"
                                                    name="partyLogo"
                                                    placeholder="https://imagelogourl.com/image.jpg"
                                                    accept="image/gif, image/png, image/jpeg"
                                                    id="partyLogo"
                                                    onChange={this.onChange}
                                                    value={this.state.partyLogo}
                                                />
                                            </div>


                                            <div className="input-group">
                                                <label>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</label>
                                                <button
                                                    type="submit"
                                                    name="submit"
                                                    className="btn"
                                                    id="createPartyBtn"
                                                >Create Party</button>
                                            </div>

                                        </form>
                                    </div>
                                </div>
                            </div>
                        </div>
                    )}


                    <main className="cards">
                        {parties && parties.map((party) => {
                            return (
                                <Fragment key={party.id}>
                                    <PartyCard image={party.logourl ? party.logourl : apcLogo} name={party.name} alias={party.alias} address={party.hqaddress} />
                                </Fragment>
                            );
                        })}

                    </main>
                </div>
                <Footer />
            </div>

        );
    }
}

const mapDispatchToProps = {
    listParties: getParty,
    createNewParty: createParty
};

const mapStateToProps = (state) => ({
    parties: state.party.party,
    createPartyError: state.party.createPartyError,
    status: state.party.status,
    errorMessage: state.party.error
});
export default connect(mapStateToProps, mapDispatchToProps)(PartiesAdmin);

