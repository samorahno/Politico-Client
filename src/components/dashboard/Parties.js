import React from 'react';
import '../../styles/dashboard.css';
import Header from '../nav/DashboardHeader';
import Footer from '../nav/DashboardFooter';
import SideBar from '../nav/DashboardSidebar';

const Parties = () => {
    return (
        <div>
            <Header />
            <SideBar />
            <div className="main-content">
                <div className="title">
					home /
                </div>

                <div className="main">

                    <div className="widget widget_table" style={{ overflow: 'auto' }}>
                        <div className="title heado">Political Parties List</div>

                        <div className="widget_stat widget_stat_table">
                            <div style={{ overflow: 'auto' }}>
                                <table style={{ overflow: 'scroll' }}>
                                    <thead>
                                        <tr>
                                            <th>S/N</th>
                                            <th>Name</th>
                                            <th>Head Quarters</th>
                                            <th>Action</th>
                                        </tr>
                                    </thead>

                                    <tbody id="partyContainer">
                                        <tr>
                                            <td>A</td>
                                            <td>B</td>
                                            <td>C</td>
                                            <td>D</td>
                                        </tr>
                                        <tr>
                                            <td>A</td>
                                            <td>B</td>
                                            <td>C</td>
                                            <td>D</td>
                                        </tr>
                                    </tbody>

                                </table>
                            </div>
                        </div>
                    </div>

                </div>
            </div>
            <Footer />
        </div>

    );
};

export default Parties;
