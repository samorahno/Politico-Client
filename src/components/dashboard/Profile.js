import React from 'react';
import '../../styles/dashboard.css';
import Header from '../nav/DashboardHeader';
import Footer from '../nav/DashboardFooter';
import SideBar from '../nav/DashboardSidebar';
import defaultPix from '../../images/Anonymous.jpg';

const Profile = () => {
    return (
        <div>
            <Header />
            <SideBar />
            <div className="main-content">
                <div className="title" />

                <div className="main">
                    <div className="widget widget_table" style={{ overflow: 'auto' }}>
                        <div className="title heado">User Profile</div>
                        <div className="widget_stat widget_stat_table">
                            <div style={{ overflow: 'auto' }}>
                                <table style={{ overflow: 'scroll' }}>
                                    <tr>
                                        <td colSpan="2"><img src={defaultPix} alt="User Image" className="userimage" /></td>

                                    </tr>

                                    <tr>
                                        <td>Name</td>
                                        <td>Kikelomo Folagbade</td>
                                    </tr>

                                    <tr>
                                        <td>Email</td>
                                        <td>samabos007@gmail.com</td>
                                    </tr>
                                    <tr>
                                        <td>Phone</td>
                                        <td>07036463311</td>
                                    </tr>
                                    <tr>
                                        <td>Joined On</td>
                                        <td>20 - 01 - 2019</td>
                                    </tr>

                                </table>
                            </div>

                        </div>

                    </div>
                </div>

                <div className="main votesize" >
                    <div className="widget">
                        <div className="title heado">Vote Count</div>
                        <div className="widget_stat">
                            <div className="widget_stat1">
                                <i className="fa  fa-check-circle-o" />
                            </div>
                            <div className="widget_stat2">
						3
                            </div>
                        </div>
                    </div>

                </div>


                <div className="main">
                    <div className="widget widget_table" style={{ overflow: 'auto' }}>
                        <div className="title heado">Vote History</div>
                        <div className="widget_stat widget_stat_table">
                            <div className="overflo">
                                <table>
                                    <tr>
                                        <th>S/N</th>
                                        <th>Voted for</th>
                                        <th>Party</th>
                                        <th>Office</th>
                                        <th>Date</th>

                                    </tr>

                                    <tr >
                                        <td>1</td>
                                        <td>Wale Ojo</td>
                                        <td>PDP</td>
                                        <td>National Assembly Member</td>
                                        <td>23-01-2019</td>
                                    </tr>

                                    <tr>
                                        <td>2</td>
                                        <td>Tola Odeyemi</td>
                                        <td>PDP</td>
                                        <td>Legislative Council Member</td>
                                        <td>23-01-2019</td>
                                    </tr>

                                    <tr >
                                        <td>3</td>
                                        <td>Goke Talabi</td>
                                        <td>PDP</td>
                                        <td>Local Government Chairman</td>
                                        <td>23-01-2019</td>
                                    </tr>


                                </table>
                            </div>

                        </div>

                    </div>
                </div>


            </div>
            <Footer />
        </div>

    );
};

export default Profile;
