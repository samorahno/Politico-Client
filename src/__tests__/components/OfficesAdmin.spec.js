import React from 'react';
import { shallow } from 'enzyme';
import { OfficeAdmin } from '../../components/dashboard/Offices-Admin';

const props = {
    listOffices: jest.fn(),
    createOfficeError: '',
    status: 200,
    errorMessage: '',
    createNewOffice: jest.fn(),
    isAdmin: true,
    offices: [],
    previousProps: {
        createOfficeError: ''
    }
};

const propsError = {
    listOffices: jest.fn(),
    createOfficeError: 'nnn',
    status: 400,
    errorMessage: 'gg',
    createNewOffice: jest.fn(),
    previousProps: {
        createOfficeError: 'hshjshjs'
    }
};

const officeData = {
    name: 'office',
    type: 'federal'
};

describe('Header', () => {
    it('should render OfficeAdmin component correctly', () => {
        const component = shallow(<OfficeAdmin {...props}/>);
        expect(component).toMatchSnapshot();
    });

    it('should simulate onchange event for office name text box', () => {
        const component = shallow(<OfficeAdmin {...props}/>);
        component.find('input').at(0).simulate('change', {
            target: { value: officeData.name }
        });
    });

    it('should simulate onchange event for office type drop down menu', () => {
        const component = shallow(<OfficeAdmin {...props}/>);
        component.find('select').at(0).simulate('change', {
            target: { value: officeData.type }
        });
    });

    it('', () => {
        const component = shallow(<OfficeAdmin {...props}/>);
        component.find('form').simulate('submit', {
            preventDefault: () => {}
        });
        component.setState({
            name: officeData.name,
            type: officeData.type
        });
        expect(component).toMatchSnapshot();
    });

    it('should simulate onchange event for office type drop down menu', () => {
        const component = shallow(<OfficeAdmin {...propsError}/>);
        expect(component).toMatchSnapshot();
    });
});

