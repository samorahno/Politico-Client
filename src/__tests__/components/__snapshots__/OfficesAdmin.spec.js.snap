// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header  1`] = `ShallowWrapper {}`;

exports[`Header should render OfficeAdmin component correctly 1`] = `ShallowWrapper {}`;

exports[`Header should simulate onchange event for office type drop down menu 1`] = `ShallowWrapper {}`;
